### PR TITLE
Use mock v1alpha1 server in Beacon API's validator functions

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Cleanup + fixes

**What does this PR do? Why is it needed?**

There are 4 `ProduceXXX` methods defined on the Beacon API's validator server. Each of them gets the block to propose from a v1alpha1 server, which is a field of the validator server. Because preparing a proper block is very difficult to set up, the tests are very long and hard to read. The v1alpha1 server and its interface are autogenerated and we use `mockgen` to generate a mock for the interface. Therefore we can replace the type of the v1alpha1 server field from a concrete type to its interface type, and use the mock in tests.

To do this, we have to include `AssignValidatorToSubnet` and `AggregatedSigAndAggregationBits` in the v1alpha1 server's proto definition, so that these methods are available through the interface. We also can no longer use v1alpha1 server's exported fields directly in the validator server because they will be no longer available. Thus they have to be made fields of the validator server.

This PR also fixes some issues in the implementation of `ProduceXXX`:
- `ProduceBlockV2` and `ProduceBlockV2SSZ` didn't check for optimistic status
- `ProduceBlockV2` and `ProduceBlockV2SSZ` didn't handle non-blinded blocks coming from the v1alpha1 server
- `ProduceBlindedBlockSSZ` produced full (!) Bellatrix and Capella blocks

### CODE REVIEW

File: beacon-chain/rpc/eth/validator/validator.go [hunk 1]
```diff
@@ -349,6 +349,17 @@ func (vs *Server) ProduceBlockV2(ctx context.Context, req *ethpbv1.ProduceBlockR
 			},
 		}, nil
 	}
+	optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:352]: We should not be proposing blocks when optimistically synced, right?

File: beacon-chain/rpc/eth/validator/validator.go [hunk 2]
```diff
@@ -628,9 +660,20 @@ func (vs *Server) ProduceBlindedBlockSSZ(ctx context.Context, req *ethpbv1.Produ
...
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
+	}
+	if optimistic {
+		return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
+	}
+	_, ok = v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_Bellatrix)
+	if ok {
+		return nil, status.Error(codes.Internal, "Prepared beacon block is not blinded")
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/validator.go:672]: might be nice to specify the fork type here

File: beacon-chain/rpc/eth/validator/validator.go [hunk 3]
```diff
@@ -349,6 +349,17 @@ func (vs *Server) ProduceBlockV2(ctx context.Context, req *ethpbv1.ProduceBlockR
...
+	optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
+	}
+	if optimistic {
+		return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
+	}
+	_, ok = v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		return nil, status.Error(codes.Internal, "Prepared beacon block is blinded")
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/validator.go:361]: more comments on maybe we can specify the fork type.

File: beacon-chain/rpc/eth/validator/validator.go [hunk 4]
```diff
@@ -437,6 +452,17 @@ func (vs *Server) ProduceBlockV2SSZ(ctx context.Context, req *ethpbv1.ProduceBlo
...
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/validator.go:464]: here too

File: beacon-chain/rpc/eth/validator/validator.go [hunk 5]
```diff
@@ -671,7 +718,7 @@ func (vs *Server) PrepareBeaconProposer(
 	var validatorIndices []primitives.ValidatorIndex
 	newRecipients := make([]*ethpbv1.PrepareBeaconProposerRequest_FeeRecipientContainer, 0, len(request.Recipients))
 	for _, r := range request.Recipients {
-		f, err := vs.V1Alpha1Server.BeaconDB.FeeRecipientByValidatorID(ctx, r.ValidatorIndex)
+		f, err := vs.BeaconDB.FeeRecipientByValidatorID(ctx, r.ValidatorIndex)
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/validator.go:721]: think we should have a check if beacondb is nil here? probably missed before.

Comment (rkapka) [reply beacon-chain/rpc/eth/validator/validator.go:721]: I think it's unnecessary. This should always be initialized.

**Review by james-prysm — APPROVED** [review]

**Pull request merged**